Add club.zbridge.zbridge

amitter84 opened the pull request:
Submission of zBridge Club app. Build verified offline with node-sources.json.

*Note for maintainers: I apologize for the previous messy PR (#7343). I had failed to correctly populate the template and had some trial-and-error commits in the history. I have now cleaned up the history into a single intentional commit and completed the submission checklist below.*

### Please confirm your submission meets all the criteria

- [x] Please describe the application briefly. 
  Contract bridge is a trick-taking card game for four players, divided into two partnerships sitting opposite each other (North-South vs. East-West). zBridge is an online bridge club where you can learn and play bridge. It is a full bridge platform with multiple features: play with Jethro (a competitive bridge bot for 2/1 game forcing), try interactive tutorials, join multiplayer games in social and rated rooms, or analyse custom/random deals in the analysis room to practice different bidding and playing scenarios with hints from the AI.
- [X] Please attach a video showcasing the application on Linux using the Flatpak. 
  https://youtu.be/tibrdEAvieI
- [X] The Flatpak ID follows all the rules listed in the [Application ID requirements][appid].
- [X] I have read and followed all the [Submission requirements][reqs] and the [Submission guide][reqs2] and I agree to them.
- [X] I am a developer of the project.

[appid]: https://docs.flathub.org/docs/for-app-authors/requirements#application-id
[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[reqs2]: https://docs.flathub.org/docs/for-app-authors/submission

amitter84 commented:
bot, build

amitter84 commented:
bot, build

amitter84 commented:
bot, build

amitter84 commented:
bot, build

hfiguiere commented on club.zbridge.zbridge.metainfo.xml:
```diff
@@ -0,0 +1,44 @@
+  <screenshots>
+    <screenshot type="default">
+      <caption>The game table in action</caption>
+      <image>https://raw.githubusercontent.com/amitter84/zbridge-flatpak-source/main/screenshots/screenshot1.jpg</image>
```
URL should NEVER reference the tip of a branch. It's not immutable.

hfiguiere commented on icon.png:
This file should be (submitted and managed) upstream.

amitter84 commented:
bot, build

hfiguiere commented on club.zbridge.zbridge.json:
```diff
@@ -0,0 +1,2346 @@
+                "mkdir -p flatpak-node/cache/electron",
+                "printf \"2346338e2ffa12276f5b3bc6d63ff357c39f42d260ad45f19471ef7bbf18a389 *electron-v29.4.6-linux-x64.zip\\n9007e44a5cb5b7645455e7895d2039d3264cab2eb0ecf8dbb4295344d392ca74 *electron-v29.4.6-linux-armv7l.zip\\n14d5b54ef561cea2ca28c55a877bc1a1a3e360ca2f9e69ab3aa719054eca8552 *electron-v29.4.6-linux-arm64.zip\\n\" > flatpak-node/cache/electron/SHASUMS256.txt-29.4.6",
+                "npm install --offline --cache=npm-cache --ignore-scripts",
+                "chmod +x node_modules/app-builder-bin/linux/x64/app-builder",
```
this will work great on aarch64.... #sarcasm

amitter84 commented:
bot, build

amitter84 commented:
bot, build

amitter84 commented:
I've converted the manifest to YAML, separated the node dependencies into 
node-sources.json
 (cleaned of forbidden data URLs/shell types), and moved the wrapper script to a dedicated file. Build is now passing for both x86_64 and aarch64.

hfiguiere commented on club.zbridge.zbridge.metainfo.xml:
not resolved

hfiguiere commented on zbridge-club.sh:
electron should use zypak instead of disabling the sandbox

hfiguiere commented on club.zbridge.zbridge.yaml:
```diff
@@ -0,0 +1,75 @@
+- --socket=wayland
+- --socket=pulseaudio
+- --share=network
+- --device=dri
```
```suggestion
```

amitter84 commented:
@hfiguiere Switched to zypak-wrapper for the Electron sandbox.
Removed the duplicate --device=dri entry.
Updated the source tag to v1.0.4 and moved the metadata files (
.desktop
, 
.metainfo.xml
, icon) to be managed upstream, removing them from the Flathub repo.

amitter84 commented:
Kindly let me know if there is anything pending at my end or any mistake that I have overlooked.
